### PR TITLE
Rename CanCooperation to RemoteControl

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**

### CODE REVIEW

File: src/components/functional_module/test/include/mock_application.h [hunk 1]
```diff
@@ -29,8 +29,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef TEST_COMPONENTS_CAN_COOPERATION_INCLUDE_MOCK_APPLICATION_H_
-#define TEST_COMPONENTS_CAN_COOPERATION_INCLUDE_MOCK_APPLICATION_H_
+#ifndef TEST_COMPONENTS_REVERSE_CONTROL_INCLUDE_MOCK_APPLICATION_H_
```
Comment (AByzhynar) [inline src/components/functional_module/test/include/mock_application.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/functional_module/test/include/mock_service.h [hunk 2]
```diff
@@ -29,8 +29,8 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  */
-#ifndef TEST_COMPONENTS_CAN_COOPERATION_INCLUDE_MOCK_SERVICE_H_
-#define TEST_COMPONENTS_CAN_COOPERATION_INCLUDE_MOCK_SERVICE_H_
+#ifndef TEST_COMPONENTS_REVERSE_CONTROL_INCLUDE_MOCK_SERVICE_H_
+#define TEST_COMPONENTS_REVERSE_CONTROL_INCLUDE_MOCK_SERVICE_H_
```
Comment (AByzhynar) [inline src/components/functional_module/test/include/mock_service.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/CMakeLists.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-set(target "CANCooperation")
+set(target "ReverseControllModule")
```
Comment (AByzhynar) [inline src/components/reverse_control/CMakeLists.txt]: @LuxoftAKutsan Control with 1 `l` letter

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/reverse_control/include/reverse_control/event_engine/event.h [hunk 4]
```diff
@@ -30,15 +30,15 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_EVENT_ENGINE_CAN_COOPERATION_EVENT_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_EVENT_ENGINE_CAN_COOPERATION_EVENT_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_EVENT_ENGINE_REVERSE_CONTROL_EVENT_H_
+#define SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_EVENT_ENGINE_REVERSE_CONTROL_EVENT_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/event_engine/event.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/include/reverse_control/rc_command_factory.h [hunk 5]
```diff
@@ -30,16 +30,16 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_MOBILE_COMMAND_FACTORY_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_MOBILE_COMMAND_FACTORY_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_MOBILE_COMMAND_FACTORY_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/rc_command_factory.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/include/reverse_control/rc_module_constants.h [hunk 6]
```diff
@@ -30,10 +30,10 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_CONSTANTS_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_CONSTANTS_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_CONSTANTS_H_
+#define SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_CONSTANTS_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/rc_module_constants.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/include/reverse_control/rc_module_timer.h [hunk 7]
```diff
@@ -30,12 +30,12 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_MODULE_INCLUDE_CAN_MODULE_CAN_MODULE_TIMER_H_
-#define SRC_COMPONENTS_CAN_MODULE_INCLUDE_CAN_MODULE_CAN_MODULE_TIMER_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_PLUGIN_INCLUDE_REVERSE_CONTROL_PLUGIN_RC_MODULE_TIMER_H_
+#define SRC_COMPONENTS_REVERSE_CONTROL_PLUGIN_INCLUDE_REVERSE_CONTROL_PLUGIN_RC_MODULE_TIMER_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/rc_module_timer.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/include/reverse_control/reverse_control_event.h [hunk 8]
```diff
@@ -30,34 +30,34 @@
  POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_EVENT_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_EVENT_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_EVENT_H_
+#define SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_EVENT_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/reverse_control_event.h]: @LuxoftAKutsan Please edit header guard as it's wrong

File: src/components/reverse_control/include/reverse_control/reverse_control_plugin.h [hunk 9]
```diff
@@ -30,24 +30,24 @@
...
 
-namespace can_cooperation {
+namespace reverse_control {
 
-class CANModule : public CANModuleInterface {
+class ReverseControllPlugin : public ReversePluginInterface {
  public:
-  CANModule();
-  ~CANModule();
+  ReverseControllPlugin();
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/reverse_control_plugin.h]: @LuxoftAKutsan Redundant letter `l`

File: src/components/reverse_control/include/reverse_control/reverse_plugin_interface.h [hunk 10]
```diff
@@ -30,26 +30,26 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_INTERFACE_H_
-#define SRC_COMPONENTS_CAN_COOPERATION_INCLUDE_CAN_COOPERATION_CAN_MODULE_INTERFACE_H_
+#ifndef SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_INTERFACE_H_
+#define SRC_COMPONENTS_REVERSE_CONTROL_INCLUDE_REVERSE_CONTROL_REVERSE_CONTROL_PLUGIN_INTERFACE_H_
```
Comment (AByzhynar) [inline src/components/reverse_control/include/reverse_control/reverse_plugin_interface.h]: @LuxoftAKutsan Please edit header guard as it's wrong

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/reverse_control/src/reverse_control_plugin.cc [hunk 11]
```diff
@@ -290,7 +285,7 @@ void CANModule::RemoveAppExtensions() {
   }
 }
 
-void CANModule::RemoveAppExtension(uint32_t app_id) {
+void ReverseControllPlugin::RemoveAppExtension(uint32_t app_id) {
```
Comment (AByzhynar) [inline src/components/reverse_control/src/reverse_control_plugin.cc]: @LuxoftAKutsan Double `l`

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/reverse_control/test/CMakeLists.txt [hunk 12]
```diff
@@ -16,17 +16,16 @@ include_directories (
...
+  src/rc_app_extension_test.cc
   ${CMAKE_SOURCE_DIR}/src/components/application_manager/test/mock_message_helper.cc
 )
 
 set (LIBRARIES
   gtest
   gmock
   gmock_main
-  CANCooperation
+  ReverseControllModule
```
Comment (AByzhynar) [inline src/components/reverse_control/test/CMakeLists.txt]: @LuxoftAKutsan Double L

File: src/components/reverse_control/test/commands/CMakeLists.txt [hunk 13]
```diff
@@ -32,24 +32,22 @@ include(${CMAKE_SOURCE_DIR}/tools/cmake/helpers/sources.cmake)
...
   ${COMPONENTS_DIR}/application_manager/test/mock_message_helper.cc
-#  ${COMPONENTS_DIR}/application_manager/src/smart_object_keys.cc
   ${RC_COMMANDS_TEST_DIR}/*
-  ${COMPONENTS_DIR}/can_cooperation/test/commands/get_interior_vehicle_data_request_test.cc
 )
 
 set(LIBRARIES
   gmock
-  CANCooperation
+  ReverseControllModule
```
Comment (AByzhynar) [inline src/components/reverse_control/test/commands/CMakeLists.txt]: @LuxoftAKutsan Double L

File: src/components/functional_module/test/include/mock_application.h (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/functional_module/test/include/mock_application.h]: fixed

File: src/components/functional_module/test/include/mock_service.h (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/functional_module/test/include/mock_service.h]: fixed

File: src/components/reverse_control/include/reverse_control/event_engine/event.h (hunk 4 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/include/reverse_control/event_engine/event.h]: fixed

File: src/components/reverse_control/include/reverse_control/rc_module_constants.h (hunk 6 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/include/reverse_control/rc_module_constants.h]: fixed

File: src/components/reverse_control/include/reverse_control/reverse_control_event.h (hunk 8 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/include/reverse_control/reverse_control_event.h]: fixed

File: src/components/reverse_control/include/reverse_control/reverse_plugin_interface.h (hunk 10 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/include/reverse_control/reverse_plugin_interface.h]: fixed

File: src/components/reverse_control/src/reverse_control_plugin.cc (hunk 11 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/src/reverse_control_plugin.cc]: fixed

File: src/components/reverse_control/test/commands/CMakeLists.txt (hunk 13 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/test/commands/CMakeLists.txt]: fixed

File: src/components/reverse_control/test/CMakeLists.txt (hunk 12 above)
Comment (LuxoftAKutsan) [reply src/components/reverse_control/test/CMakeLists.txt]: fixed

**Comment by LuxoftAKutsan** [comment]
See #174

**Pull request closed without merging**